### PR TITLE
added filter/date/valueExpression

### PR DESCRIPTION
**Pull request by fceller**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -89,24 +92,54 @@ router
...
+    ? `FILTER ${filterExpression}`
+    : "");
+
+  let dateSnippet = aql.literal(dateExpression
+    ? `LET d = ${dateExpression}`
+    : `LET d = doc["${dateField}"]`);
+
+  let valueSnippet = aql.literal(valueExpression
+    ? `LET v = ${valueExpression}`
+    : `LET v = doc["${valueField}"]`);
```
Comment (pluma) [inline index.js]: FWIW This allows entering arbitrary AQL, including data modification queries.

That said, I'd suggest doing it this way:

```js
let valueSnippet = aql`LET v = ${
  valueExpression
  ? aql.literal(valueExpression)
  : aql`doc[${valueField}]`
}`;
```

Also note that ```
aql``
``` is equivalent to `aql.literal("")`.

Comment (pluma) [reply index.js]: Using the `aql` template tag for the value field avoids problems with weird field names.

File: manifest.json [hunk 2]
```diff
@@ -22,18 +22,33 @@
...
     },
+    "filterExpression": {
+      "type": "string",
+      "description": "An AQL expression used to filter data. The current document is called 'doc'.",
+      "required": false
+    },
     "dateField": {
       "type": "string",
-      "description": "Name of the field containing the date time for each data point.",
+      "description": "Name of the field containing the date time for each data point. Either set 'dateField' or 'dateExpression'.",
```
Comment (pluma) [inline manifest.json]: I'd elaborate that the `*Expression` configs take precedence if both are specified.

File: index.js [hunk 3]
```diff
@@ -89,24 +92,54 @@ router
...
+
+  require("internal").print(`
```
Comment (pluma) [inline index.js]: This should probably be removed if this is no longer WIP.

File: index.js [hunk 4]
```diff
@@ -14,37 +14,40 @@ const { getAuth } = require("./util");
...
-) {
+const AGG_NAME = cfg.aggregation.toUpperCase();
+
+if (!allowAggregations.includes(AGG_NAME)) {
+  const allowed = allowAggregations.join(", ");
   throw new Error(
     `Invalid service configuration. Unknown aggregation function: ${
       cfg.aggregation
-    }`
+    }, allow are ${allowed}`
```
Comment (pluma) [inline index.js]: I think this is meant to read "allowed are" but in any case just inlining the array results in icky formatting.

Maybe try this:
```js
`Invalid service configuration. Unknown aggregation function: ${
  cfg.aggregation
}. Please use one of: ${allowed.join(", ")}.`
```

**Review by pluma — APPROVED** [review]

**Review by pluma — CHANGES_REQUESTED** [review]
approved by mistake, sorry

File: index.js (hunk 1 above)
Comment (fceller) [reply index.js]: done

File: manifest.json (hunk 2 above)
Comment (fceller) [reply manifest.json]: done

File: index.js (hunk 3 above)
Comment (fceller) [reply index.js]: done

File: index.js (hunk 4 above)
Comment (fceller) [reply index.js]: done

**Comment by fceller** [comment]
@pluma please review the changes, thx

**Pull request merged**